cajita/unit_test/tstLocalGrid.hpp

```diff
@@ -27,6 +27,192 @@ using namespace Cajita;
+        EXPECT_EQ( boundary_space.max( d ),
+                   owned_space.max( d ) + max_upper_shift );
+    }
+}
+
+template <class BoundaryType, class OwnedType, std::size_t NSD>
+void checkCellBoundary( BoundaryType boundary_space, OwnedType owned_space,
+                        const std::array<int, NSD> neighbor,
+                        const int min_lower_shift, const int min_upper_shift,
+                        const int max_lower_shift, const int max_upper_shift )
+{
+    for ( std::size_t d = 0; d < NSD; ++d )
+        checkBoundary( boundary_space, owned_space, neighbor, min_lower_shift,
+                       min_upper_shift, max_lower_shift, max_upper_shift, d );
+}
+
+template <class BoundaryType, class OwnedType, std::size_t NSD>
+void checkNodeBoundary( BoundaryType boundary_space, OwnedType owned_space,
+                        const std::array<int, NSD> neighbor,
+                        const int min_lower_shift, const int min_upper_shift,
+                        const int max_lower_shift, const int max_upper_shift )
+{
+    for ( std::size_t d = 0; d < NSD; ++d )
+        checkBoundary( boundary_space, owned_space, neighbor, min_lower_shift,
+                       min_upper_shift - 1, max_lower_shift + 1,
+                       max_upper_shift, d );
+}
+
+//---------------------------------------------------------------------------//
+template <class BoundaryType, class OwnedType, std::size_t NSD>
+void checkFaceBoundary( BoundaryType boundary_space, OwnedType owned_space,
+                        const std::array<int, NSD> neighbor,
+                        const int min_lower_shift, const int min_upper_shift,
+                        const int max_lower_shift, const int max_upper_shift,
+                        const std::size_t Dir )
+{
+    for ( std::size_t d = 0; d < NSD; ++d )
+    {
+        int min_upper_shift_dir = 0;
+        int max_lower_shift_dir = 0;
+        if ( Dir == d )
+        {
+            min_upper_shift_dir = -1;
+            max_lower_shift_dir = 1;
+        }
+
+        checkBoundary( boundary_space, owned_space, neighbor, min_lower_shift,
+                       min_upper_shift + min_upper_shift_dir,
+                       max_lower_shift + max_lower_shift_dir, max_upper_shift,
+                       d );
+    }
+}
+
+//---------------------------------------------------------------------------//
+template <class BoundaryType, class OwnedType, std::size_t NSD>
+void checkEdgeBoundary( BoundaryType boundary_space, OwnedType owned_space,
+                        const std::array<int, NSD> neighbor,
+                        const int min_lower_shift, const int min_upper_shift,
+                        const int max_lower_shift, const int max_upper_shift,
+                        const std::size_t Dir )
+{
+    for ( std::size_t d = 0; d < NSD; ++d )
+    {
+        int min_upper_shift_dir = 0;
+        int max_lower_shift_dir = 0;
+        if ( Dir != d )
```

Comment:
Am I missing something here? I would expect edges to end up the same as faces

Reply:
`edge` behave differently when `Dir == d`, it behave like a `Cell`. But `face` behave like a `Node` when `Dir == d`